tests/sentry/flags/endpoints/test_hooks.py

```diff
@@ -1,26 +1,224 @@
+from urllib.parse import quote
+
+import pytest
 from django.urls import reverse
+from rest_framework.exceptions import AuthenticationFailed
 
+from sentry.flags.endpoints.hooks import get_org_id_from_token
+from sentry.flags.models import ACTION_MAP, CREATED_BY_TYPE_MAP, FlagAuditLogModel
+from sentry.models.orgauthtoken import OrgAuthToken
+from sentry.silo.base import SiloMode
 from sentry.testutils.cases import APITestCase
+from sentry.testutils.silo import assume_test_silo_mode
 from sentry.utils.security.orgauthtoken_token import hash_token
 
 
 class OrganizationFlagsHooksEndpointTestCase(APITestCase):
-    endpoint = "sentry-api-0-organization-flag-hooks"
-    provider = "test"
+    endpoint = "sentry-api-0-flag-hooks"
 
     def setUp(self):
         super().setUp()
-        self.url = reverse(self.endpoint, args=(self.organization.slug, self.provider))
-
-    def test_post(self):
-        token = "sntrys_abc123_xyz"
-        self.create_org_auth_token(
-            name="Test Token 1",
-            token_hashed=hash_token(token),
-            organization_id=self.organization.id,
-            token_last_characters="xyz",
-            scope_list=["org:ci"],
-            date_last_used=None,
-        )
-
-        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer {token}")
+
+    def test_bad_token(self):
+        token_str = "badtoken"
+        url = reverse(self.endpoint, args=("launchdarkly", token_str))
+
+        response = self.client.post(url, {})
+
+        assert response.status_code == 403
+
+    def test_bad_org(self):
+        with assume_test_silo_mode(SiloMode.CONTROL):
+            token_str = "sntrys+_abc123_xyz"
+            token_encoded = quote(token_str)
+            OrgAuthToken.objects.create(
+                name="Test Token 1",
+                token_hashed=hash_token(token_str),
+                organization_id=1234,
+                token_last_characters="xyz",
+                scope_list=["org:ci"],
+                date_last_used=None,
+            )
+        with pytest.raises(AuthenticationFailed):
+            get_org_id_from_token(token_encoded)
```

Comment:
Since this is not explicitly targeting the HTTP endpoint let's move it to a different class not called `OrganizationFlagsHooksEndpointTestCase`.

Reply:
That applies for any unit test in this class.